Learn content

wackerow opened the pull request:
## Description
- Installs basic markdown support with `react-markdown`
- ~~Fetches further reading and tutorials resources from https://raw.githubusercontent.com/rkdud007/awesome-zkvm/refs/heads/main/README.md~~
- ~~Replaces those two sections with fetched data to make that list more maintainable~~
- Add /learn page content in markdown format to `app/learn/content/index.md`
- Error thrown if file doesn't exist, else renders parses with gray matter and rendered with react markdown.
- Markdown Provider added to `components/Markdown/Provider.tsx`
- Markdown Components added to `components/Markdown/Components.tsx`

cc: @rkdud007

## Preview link
https://deploy-preview-150--ethproofs.netlify.app/learn

wackerow commented on app/learn/page.tsx:
```diff
@@ -1,105 +1,402 @@
+            "opacity-80 contrast-[110%] hue-rotate-180 invert", // Light mode filters
+            "dark:opacity-100 dark:contrast-100 dark:hue-rotate-0 dark:invert-0" // Dark mode filter resets
+          )}
+          alt=""
```
```suggestion
          alt=""
          priority
```

rkdud007 commented:
is it possible to have table https://github.com/rkdud007/awesome-zkvm?tab=readme-ov-file#projects on here? 
maybe to easily maintain data if theres way to mirror the readme directly on frontend would be nice (not required)

Maybe it should be zkVM not zkEVM I veiew ethproofs more like proof explorer for block execution generated via various zkVMs

wackerow commented:
> is it possible to have table https://github.com/rkdud007/awesome-zkvm?tab=readme-ov-file#projects on here? maybe to easily maintain data if theres way to mirror the readme directly on frontend would be nice (not required)
> 
> Maybe it should be zkVM not zkEVM I veiew ethproofs more like proof explorer for block execution generated via various zkVMs

Yeah I think that would be a nice upgrade. We could implement markdown support and fetch https://raw.githubusercontent.com/rkdud007/awesome-zkvm/refs/heads/main/README.md to parse for that header section then style and render it. I'm just not sure I see this included in the initial site launch, but for a follow-up.

pettinarip reviewed (commented):
seems that the title is not parsing correctly in this case
![image](https://github.com/user-attachments/assets/981b4a49-379f-4d06-b409-f28670d797b4)

wackerow commented:
> seems that the title is not parsing correctly in this case
> 
> ![image](https://github.com/user-attachments/assets/981b4a49-379f-4d06-b409-f28670d797b4)
> 
> 

ok, I'm not sure this added fetch is worth it. I think a more reliable approach is to simply import what we need locally to this repo. can keep in markdown for now for  easier editing in the future, but the fetch is an unnecessary step that could cause issues. will patch

wackerow commented:
Updated description, but after the basic markdown support was installed, didn't make much sense to fetch just those two sections and inject it into a react page. 

I refactored this as a simple markdown content page, which we can then render. Removes dependency on the `fetch`, and keeps all content for the page readable and easy to edit.

pettinarip reviewed (approved):
LGTM
